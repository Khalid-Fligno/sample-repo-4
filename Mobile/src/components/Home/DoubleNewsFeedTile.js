import React from "react";
import {
  Dimensions,
  TouchableOpacity,
  Text,
  StyleSheet,
  ImageBackground,
  View,
  Animated,
} from "react-native";
import PropTypes from "prop-types";
import colors from "../../styles/colors";
import fonts from "../../styles/fonts";

const { width } = Dimensions.get("window");

const DoubleNewsFeedTile = (props) => {
  const {
    imageLeft,
    imageRight,
    titleLeft1,
    titleLeft2,
    titleRight1,
    titleRight2,
    onPressLeft,
    onPressRight,
  } = props;
  const animatedValueLeft = new Animated.Value(1);
  const animatedValueRight = new Animated.Value(1);

  const animatedStyleLeft = {
    transform: [{ scale: animatedValueLeft }],
  };
  const animatedStyleRight = {
    transform: [{ scale: animatedValueRight }],
  };
  const handlePressIn = () => {
    Animated.spring(animatedValueLeft, {
      toValue: 0.92,
      useNativeDriver: true,
    }).start();
  };
  const handlePressOut = () => {
    Animated.spring(animatedValueLeft, {
      toValue: 1,
      friction: 3,
      tension: 40,
      useNativeDriver: true,
    }).start();
  };
  const handlePressInRight = () => {
    Animated.spring(animatedValueRight, {
      toValue: 0.92,
      useNativeDriver: true,
    }).start();
  };
  const handlePressOutRight = () => {
    Animated.spring(animatedValueRight, {
      toValue: 1,
      friction: 3,
      tension: 40,
      useNativeDriver: true,
    }).start();
  };

  return (
    <View style={styles.doubleTileContainer}>
      <TouchableOpacity
        delayPressIn={50}
        onPress={onPressLeft}
        style={styles.cardContainer}
        onPressIn={handlePressIn}
        onPressOut={handlePressOut}
      >
        <Animated.View style={[styles.flexContainer, animatedStyleLeft]}>
          <ImageBackground
            resizeMode="cover"
            source={imageLeft}
            style={styles.image}
          >
            <View style={styles.opacityLayer}>
              <View style={styles.titleContainer}>
                <Text style={styles.title}>{titleLeft1}</Text>
                {titleLeft2 && <Text style={styles.title}>{titleLeft2}</Text>}
              </View>
            </View>
          </ImageBackground>
        </Animated.View>
      </TouchableOpacity>
      <TouchableOpacity
        delayPressIn={60}
        onPress={onPressRight}
        style={styles.cardContainer}
        onPressIn={handlePressInRight}
        onPressOut={handlePressOutRight}
      >
        <Animated.View style={[styles.flexContainer, animatedStyleRight]}>
          <ImageBackground
            resizeMode="cover"
            source={imageRight}
            style={styles.image}
          >
            <View style={styles.opacityLayer}>
              <View style={styles.titleContainer}>
                <Text style={styles.title}>{titleRight1}</Text>
                {titleRight2 && <Text style={styles.title}>{titleRight2}</Text>}
              </View>
            </View>
          </ImageBackground>
        </Animated.View>
      </TouchableOpacity>
    </View>
  );
};

export default DoubleNewsFeedTile;

DoubleNewsFeedTile.propTypes = {
  imageLeft: PropTypes.number.isRequired,
  imageRight: PropTypes.number.isRequired,
  titleLeft1: PropTypes.string.isRequired,
  titleLeft2: PropTypes.string,
  titleRight1: PropTypes.string.isRequired,
  titleRight2: PropTypes.string,
  onPressLeft: PropTypes.func.isRequired,
  onPressRight: PropTypes.func.isRequired,
};

DoubleNewsFeedTile.defaultProps = {
  titleLeft2: null,
  titleRight2: null,
};

const styles = StyleSheet.create({
  doubleTileContainer: {
    flex: 1,
    height: (width - 30) / 2,
    flexDirection: "row",
  },
  cardContainer: {
    flex: 1,
    margin: 5,
    shadowColor: colors.charcoal.standard,
    shadowOpacity: 0.5,
    shadowOffset: { width: 0, height: 2 },
    shadowRadius: 4,
  },
  flexContainer: {
    flex: 1,
  },
  image: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    overflow: "hidden",
    borderRadius: 1,
  },
  opacityLayer: {
    flex: 1,
    width: "100%",
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: colors.transparentBlackLight,
  },
  titleContainer: {
    maxWidth: width / 2.4,
  },
  title: {
    fontFamily: fonts.bold,
    fontSize: 16,
    color: colors.white,
    textAlign: "center",
    shadowColor: colors.black,
    shadowOpacity: 1,
    shadowOffset: { width: 0, height: 0 },
    shadowRadius: 5,
  },
});
