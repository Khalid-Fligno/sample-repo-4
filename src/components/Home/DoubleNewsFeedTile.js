import React from 'react';
import {
  Dimensions,
  TouchableOpacity,
  Text,
  StyleSheet,
  ImageBackground,
  View,
  Animated,
} from 'react-native';
import PropTypes from 'prop-types';
import colors from '../../styles/colors';
import fonts from '../../styles/fonts';

const { width } = Dimensions.get('window');

export default class DoubleNewsFeedTile extends React.PureComponent {
  constructor(props) {
    super(props);
    this.animatedValueLeft = new Animated.Value(1);
    this.animatedValueRight = new Animated.Value(1);
  }
  handlePressIn = () => {
    Animated.spring(this.animatedValueLeft, {
      toValue: 0.92,
    }).start();
  }
  handlePressOut = () => {
    Animated.spring(this.animatedValueLeft, {
      toValue: 1,
      friction: 3,
      tension: 40,
    }).start();
  }
  handlePressInRight = () => {
    Animated.spring(this.animatedValueRight, {
      toValue: 0.92,
    }).start();
  }
  handlePressOutRight = () => {
    Animated.spring(this.animatedValueRight, {
      toValue: 1,
      friction: 3,
      tension: 40,
    }).start();
  }
  render() {
    const {
      imageLeft,
      imageRight,
      titleLeft1,
      titleLeft2,
      titleRight1,
      titleRight2,
      onPressLeft,
      onPressRight,
    } = this.props;
    const animatedStyleLeft = {
      transform: [{ scale: this.animatedValueLeft }],
    };
    const animatedStyleRight = {
      transform: [{ scale: this.animatedValueRight }],
    };
    return (
      <View style={styles.doubleTileContainer}>
        <TouchableOpacity
          onPress={onPressLeft}
          style={styles.cardContainer}
          onPressIn={this.handlePressIn}
          onPressOut={this.handlePressOut}
        >
          <Animated.View
            style={[{ flex: 1 }, animatedStyleLeft]}
          >
            <ImageBackground
              resizeMode="cover"
              source={imageLeft}
              style={styles.image}
            >
              <View style={styles.titleContainer}>
                <Text style={styles.title}>
                  {titleLeft1}
                </Text>
                <Text style={styles.title}>
                  {titleLeft2}
                </Text>
              </View>
            </ImageBackground>
          </Animated.View>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={onPressRight}
          style={styles.cardContainer}
          onPressIn={this.handlePressInRight}
          onPressOut={this.handlePressOutRight}
        >
          <Animated.View
            style={[{ flex: 1 }, animatedStyleRight]}
          >
            <ImageBackground
              resizeMode="cover"
              source={imageRight}
              style={styles.image}
            >
              <View style={styles.titleContainer}>
                <Text style={styles.title}>
                  {titleRight1}
                </Text>
                <Text style={styles.title}>
                  {titleRight2}
                </Text>
              </View>
            </ImageBackground>
          </Animated.View>
        </TouchableOpacity>
      </View>
    );
  }
}

DoubleNewsFeedTile.propTypes = {
  imageLeft: PropTypes.number.isRequired,
  imageRight: PropTypes.number.isRequired,
  titleLeft1: PropTypes.string.isRequired,
  titleLeft2: PropTypes.string.isRequired,
  titleRight1: PropTypes.string.isRequired,
  titleRight2: PropTypes.string.isRequired,
  onPressLeft: PropTypes.func.isRequired,
  onPressRight: PropTypes.func.isRequired,
};

const styles = StyleSheet.create({
  doubleTileContainer: {
    flex: 1,
    flexDirection: 'row',
  },
  cardContainer: {
    flex: 1,
    margin: 5,
    shadowColor: colors.charcoal.standard,
    shadowOpacity: 0.3,
    shadowOffset: { width: 0, height: 2 },
    shadowRadius: 4,
  },
  image: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    overflow: 'hidden',
    borderRadius: 1,
  },
  titleContainer: {
    backgroundColor: 'rgba(255, 255, 255, 0.6 )',
    padding: 8,
    paddingBottom: 3,
    maxWidth: width / 2.4,
    borderRadius: 2,
  },
  title: {
    fontFamily: fonts.bold,
    fontSize: 15,
    textAlign: 'center',
  },
});
