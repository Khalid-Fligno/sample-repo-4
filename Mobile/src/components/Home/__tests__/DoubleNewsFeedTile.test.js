import React from 'react';
import renderer from 'react-test-renderer';
import DoubleNewsFeedTile from '../DoubleNewsFeedTile';

describe('NewsFeedTile', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<DoubleNewsFeedTile
      imageLeft={1}
      imageRight={1}
      titleLeft1="Test Title Left"
      titleRight1="Test Title Right"
      onPressLeft={jest.fn()}
      onPressRight={jest.fn()}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
