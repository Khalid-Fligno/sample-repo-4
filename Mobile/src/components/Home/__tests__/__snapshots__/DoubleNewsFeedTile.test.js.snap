// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsFeedTile renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "height": 360,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "margin": 5,
        "opacity": 1,
        "shadowColor": "rgb(38, 38, 40)",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 4,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 1,
            "flex": 1,
            "justifyContent": "center",
            "overflow": "hidden",
          }
        }
      >
        <Image
          resizeMode="cover"
          source={1}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
              undefined,
            ]
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0, 0, 0, 0.5 )",
              "flex": 1,
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "maxWidth": 312.5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "rgb(255, 255, 255)",
                  "fontFamily": "GothamBold",
                  "fontSize": 16,
                  "shadowColor": "rgb(0, 0, 0)",
                  "shadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 5,
                  "textAlign": "center",
                }
              }
            >
              Test Title Left
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "margin": 5,
        "opacity": 1,
        "shadowColor": "rgb(38, 38, 40)",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 4,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 1,
            "flex": 1,
            "justifyContent": "center",
            "overflow": "hidden",
          }
        }
      >
        <Image
          resizeMode="cover"
          source={1}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
              undefined,
            ]
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0, 0, 0, 0.5 )",
              "flex": 1,
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "maxWidth": 312.5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "rgb(255, 255, 255)",
                  "fontFamily": "GothamBold",
                  "fontSize": 16,
                  "shadowColor": "rgb(0, 0, 0)",
                  "shadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 5,
                  "textAlign": "center",
                }
              }
            >
              Test Title Right
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
